File: backend/app/schemas/health_profile.py
from pydantic import BaseModel
from typing import Optional
import uuid
from app.models.enums import HealthCondition

class HealthProfileRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    primary_condition: HealthCondition
    has_asthma: bool
    has_copd: bool
    has_heart_disease: bool
    is_elderly: bool
    is_pregnant: bool
    is_child: bool
    
    notif_email: bool
    notif_push: bool
    notif_sms: bool
    notif_daily_summary: bool
    notif_health_alerts: bool
    notif_route_recommendations: bool
    notif_weekly_summary: bool
    notif_emergency_only: bool

    model_config = {"from_attributes": True}

class HealthProfileUpdate(BaseModel):
    primary_condition: Optional[HealthCondition] = None
    has_asthma: Optional[bool] = None
    has_copd: Optional[bool] = None
    has_heart_disease: Optional[bool] = None
    is_elderly: Optional[bool] = None
    is_pregnant: Optional[bool] = None
    is_child: Optional[bool] = None

    notif_email: Optional[bool] = None
    notif_push: Optional[bool] = None
    notif_sms: Optional[bool] = None
    notif_daily_summary: Optional[bool] = None
    notif_health_alerts: Optional[bool] = None
    notif_route_recommendations: Optional[bool] = None
    notif_weekly_summary: Optional[bool] = None
    notif_emergency_only: Optional[bool] = None

File: frontend/src/features/health/validation/healthProfile.ts
import { z } from 'zod';

const healthConditionEnum = z.enum(['none', 'asthma', 'copd', 'heart_disease', 'elderly', 'children', 'pregnant']);

export const healthProfileSchema = z.object({
  primary_condition: healthConditionEnum.default('none'),
  has_asthma: z.boolean().default(false),
  has_copd: z.boolean().default(false),
  has_heart_disease: z.boolean().default(false),
  is_elderly: z.boolean().default(false),
  is_pregnant: z.boolean().default(false),
  is_child: z.boolean().default(false),
  
  notif_email: z.boolean().default(true),
  notif_push: z.boolean().default(true),
  notif_sms: z.boolean().default(false),
  notif_daily_summary: z.boolean().default(true),
  notif_health_alerts: z.boolean().default(true),
  notif_route_recommendations: z.boolean().default(true),
  notif_weekly_summary: z.boolean().default(false),
  notif_emergency_only: z.boolean().default(false),
});

export type HealthProfileFormValues = z.infer<typeof healthProfileSchema>;
